extensions/caffeine/deployment/src/main/java/io/quarkus/caffeine/deployment/CaffeineProcessor.java

```diff
@@ -49,4 +52,15 @@ void cacheLoaders(CombinedIndexBuildItem combinedIndex, BuildProducer<Reflective
     NativeImageFeatureBuildItem nativeImageFeature() {
         return new NativeImageFeatureBuildItem(CacheConstructorsFeature.class);
     }
+
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
+    NativeImageSystemPropertyBuildItem registerRecordStatsImplementationsIfMicrometerAround(
+            MetricsCapabilityBuildItem metricsCapability) {
```

Comment:
This may help fix the native tests:
```suggestion
            Optional<MetricsCapabilityBuildItem> metricsCapability) {
```

Reply:
Ah... yes... I tested the case exercising the new code but didn't test the other modules where Micrometer is not around...